napari/layers/base/base.py

```diff
@@ -726,36 +726,41 @@ def extent(self) -> Extent:
     @property
     def _slice_indices(self):
         """(D, ) array: Slice indices in data coordinates."""
-        inv_transform = self._data_to_world.inverse
 
-        if self.ndim > self._ndisplay:
-            # Subspace spanned by non displayed dimensions
-            non_displayed_subspace = np.zeros(self.ndim)
-            for d in self._dims_not_displayed:
-                non_displayed_subspace[d] = 1
-            # Map subspace through inverse transform, ignoring translation
-            mapped_nd_subspace = inv_transform(
-                non_displayed_subspace
-            ) - inv_transform(np.zeros(self.ndim))
```

Comment:
This subtracts a second transform of zeros as a way to remove the translation. In the refactor I just generated a new Affine without any translation instead.

Most of the other lines that appear modified here are just the removal of indentation of this section